pkg/scheduler/scheduler.go

```diff
@@ -256,6 +265,47 @@ func findFlavorForResource(name corev1.ResourceName, val int64, wUsed map[string
 	return "", 0
 }
 
+func flavorSelector(spec *corev1.PodSpec, allowedKeys sets.String) nodeaffinity.RequiredNodeAffinity {
+	var specCopy corev1.PodSpec
+
+	// Remove affinity constraints with irrelevant keys.
+	if len(spec.NodeSelector) != 0 {
+		specCopy.NodeSelector = map[string]string{}
+		for k, v := range spec.NodeSelector {
+			if allowedKeys.Has(k) {
+				specCopy.NodeSelector[k] = v
+			}
+		}
+	}
+
+	affinity := spec.Affinity
+	if affinity != nil && affinity.NodeAffinity != nil && affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
+		var termsCopy []corev1.NodeSelectorTerm
+		for _, t := range affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms {
+			var expCopy []corev1.NodeSelectorRequirement
+			for _, e := range t.MatchExpressions {
+				if allowedKeys.Has(e.Key) {
+					expCopy = append(expCopy, e)
+				}
+			}
+			if len(expCopy) != 0 {
```

Comment:
isn't that the same as no term?

Reply:
if not, can you suggest a unit test (easier than explaining in detail what you have in mind :))?

Reply:
Let's say you have:

```(gpu=a10) || (non_type_label=foo)```

This should be interpreted as

```(gpu=a10) || true```

instead of

```
gpu=a10
```

Reply:
good catch, although empty term doesn't match any, but we can simply ignore all terms.

Reply:
right... once there is order support, we might want to do something else.

I thought empty expressions meant match all. We can check later.